lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,946 @@
+            plugin.apply_precip(self.fg_cube, self.precip_cube)
+
+    def test_missing_threshold_mid(self):
+        """Test that the method raises an error if the precip_cube doesn't
+        have a threshold coordinate for 7.0."""
+        self.precip_cube.coord('threshold').points = [0.5, 8., 35.]
+        plugin = Plugin()
+        msg = ("No matching high precip cube for")
+        with self.assertRaisesRegex(ConstraintMismatchError, msg):
+            plugin.apply_precip(self.fg_cube, self.precip_cube)
+
+    def test_missing_threshold_high(self):
+        """Test that the method raises an error if the precip_cube doesn't
+        have a threshold coordinate for 35.0."""
+        self.precip_cube.coord('threshold').points = [0.5, 7., 20.]
+        plugin = Plugin()
+        msg = ("No matching intense precip cube for")
+        with self.assertRaisesRegex(ConstraintMismatchError, msg):
+            plugin.apply_precip(self.fg_cube, self.precip_cube)
+
+    def test_precip_zero(self):
+        """Test that zero precip probs reduce lightning risk"""
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.0067
+        result = plugin.apply_precip(self.fg_cube, self.precip_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_precip_small(self):
+        """Test that small precip probs reduce lightning risk"""
+        self.precip_cube.data[:, 0, 7, 7] = 0.
+        self.precip_cube.data[0, 0, 7, 7] = 0.075
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.625
+        result = plugin.apply_precip(self.fg_cube, self.precip_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_precip_heavy(self):
+        """Test that prob of heavy precip increases lightning risk"""
+        self.precip_cube.data[0, 0, 7, 7] = 1.0
+        self.precip_cube.data[1, 0, 7, 7] = 0.5
+        # Set first-guess to zero
+        self.fg_cube.data[0, 7, 7] = 0.0
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.25
+        result = plugin.apply_precip(self.fg_cube, self.precip_cube)
+        self.assertArrayAlmostEqual(result.data, expected.data)
+
+    def test_precip_heavy_null(self):
+        """Test that low prob of heavy precip does not increase
+        lightning risk"""
+        self.precip_cube.data[0, 0, 7, 7] = 1.0
+        self.precip_cube.data[1, 0, 7, 7] = 0.3
+        # Set first-guess to zero
+        self.fg_cube.data[0, 7, 7] = 0.1
+        plugin = Plugin()
+        expected = set_up_cube_with_no_realizations()
+        # expected.data contains all ones except:
+        expected.data[0, 7, 7] = 0.1
```

Comment:
Don't need this with `expected = self.fg_cube.copy()` - we're expecting no change.